IPython/core/interactiveshell.py

```diff
@@ -2615,6 +2621,7 @@ def run_cell(self, raw_cell, store_history=False, silent=False, shell_futures=Tr
         result : :class:`ExecutionResult`
         """
         result = ExecutionResult()
+        self.last_execution_succeed = True
```

Comment:
I don't think a cell should be able to get this information from the `shell` once executing a cell. I'll consider it unsupported. 
This is mostly for iterm2-tools that need to change the prompt if the previous command failed. 

I'll move it though.